### PR TITLE
Centos

### PR DESCRIPTION
**Pull request by kknox**
Summary of proposed changes:
-  Provides build support for CentOS 7
-  Changes rocblas link dependency on hip to private
- rocblas definition rocblas_half changed, because gcc compilers did not understand __fp16 type
  - this is a potential backward breaking change, if users used half before

### CODE REVIEW

File: clients/include/testing_axpy.hpp [hunk 1]
```diff
@@ -84,10 +84,15 @@ rocblas_status testing_axpy(Arguments argus)
     rocblas_int N = argus.N;
     rocblas_int incx = argus.incx;
     rocblas_int incy = argus.incy;
-    T h_alpha = argus.alpha;
     rocblas_int safe_size = 100;   // arbitrarily set to 100
-
-    std::unique_ptr<rocblas_test::handle_struct> test_handle(new rocblas_test::handle_struct);
+    T h_alpha = 0;
```
Comment (amcamd) [inline clients/include/testing_axpy.hpp]: I think this needs to change to     T h_alpha = argus.alpha;

Comment (kknox) [reply clients/include/testing_axpy.hpp]: It's initialized just below, depending on T type

File: clients/include/testing_axpy.hpp [hunk 2]
```diff
@@ -124,6 +129,8 @@ rocblas_status testing_axpy(Arguments argus)
     srand(1);
     rocblas_init<T>(hx, 1, N, abs_incx);
     rocblas_init<T>(hy_1, 1, N, abs_incy);
+    // rocblas_init<T>(hx, 1, N, abs_incx, 1.0);
```
Comment (amcamd) [inline clients/include/testing_axpy.hpp]: remove these commented lines

**Comment by kknox** [comment]
I believe this may fix #143; I can now build rocblas clients with the host gcc compiler.  

Our clients still link to HiP because they use hip for memory allocation, but the rocblas library library does not force the clients to be hcc or hipcc.

**Pull request merged**